hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/rapid/RPBSPMC0Converter.java

```diff
@@ -0,0 +1,189 @@
+		mshdefaults.setReceivingFacility(exchange.getProperty(PROPERTY_SENDING_FACILITY).toString());
+		mshdefaults.setReceivingApplication(exchange.getProperty(PROPERTY_SENDING_APP).toString());
+		mshdefaults.setSendingApplication(exchange.getProperty(PROPERTY_RECEIVING_APP).toString());
+		mshdefaults.setSendingFacility(exchange.getProperty(PROPERTY_RECEIVING_FACILITY).toString());
+		mshdefaults.setUser(exchange.getProperty(PROPERTY_USER_INFO).toString());
+		mshdefaults.setMessageType(exchange.getProperty(PROPERTY_MESSAGE_TYPE).toString());
+		mshdefaults.setMessageControlId(messageDateTime);
+		return mshdefaults;
+	}
+
+	private void populateMSH(MSH msh, HL7Message mshDefaults) throws HL7Exception {
+		RapidV2MessageUtil.setMshValues(msh, mshDefaults.getSendingApplication(), mshDefaults.getSendingFacility(),
+				mshDefaults.getReceivingApplication(), mshDefaults.getReceivingFacility(), messageDateTime,
+				mshDefaults.getUser(), mshDefaults.getMessageType(),
+				StringUtils.substring(mshDefaults.getMessageControlId(), 0, 20), mshDefaults.getProcessingId());
+	}
+
+	private void populatePID(PID pid, String phn) throws HL7Exception {
+		RapidV2MessageUtil.setPidValues(pid, phn, PID_NAMESPACE_ID, PID_ID_TYPE_CODE, "", "", "");
+	}
+
+	private void populateZIA(ZIA zia, String bcResidencyDate, String surname, String firstGivenName,
+			String secondGivenName, String telephone, String immigrationOrVisaCode, String priorResidenceCode)
+			throws HL7Exception {
+
+		RapidV2MessageUtil.setZiaValues(zia, bcResidencyDate, surname, firstGivenName, secondGivenName, telephone,
+				immigrationOrVisaCode, priorResidenceCode);
+	}
+
+	private void populateMSA(MSA msa, String message) throws HL7Exception {
+		RapidV2MessageUtil.setMSAValues(msa, message);
+	}
+
+	private void populateNK1(NK1 nk1, String relationship) throws HL7Exception {
+		RapidV2MessageUtil.setNK1Values(nk1, relationship);
+	}
+
+	private void populateIN1(IN1 in1, String planEffectiveDate, String planCancellationDate, String groupNumber,
+			String groupMemberNumber, String departmentNumber) throws HL7Exception {
+		RapidV2MessageUtil.setIn1Values(in1, groupNumber, groupMemberNumber, departmentNumber, planEffectiveDate,
+				planCancellationDate);
+	}
+
+	private void populateZIH(ZIH zih, String payerCancelReason) throws HL7Exception {
+		RapidV2MessageUtil.setZihValues(zih, payerCancelReason);
+	}
+
+	private String convertDate(String date) {
+		if (date.contentEquals("0000-00-00")) {
+			return "";
+		}
+		String strDate = date;
+		DateFormat inputFormatter = new SimpleDateFormat("yyyy-MM-dd");
+		Date formattedDate = new Date();
+		try {
+			formattedDate = (Date) inputFormatter.parse(strDate);
+
+		} catch (ParseException e) {
+			logger.error("Error whhile parsing effective/cancellation date: {}", e.getMessage());
+		}
+
+		DateFormat outputFormatter = new SimpleDateFormat("yyyyMMdd");
+		String strDateTime = outputFormatter.format(formattedDate);
```

Comment:
If the exception is swallowed then the value returned is not the date provided.
The pattern can be defined as static var.
DateTimeFormatter is recommended as replacement for SimpleDateFormat since 1.8